src/modules/ekf2/EKF/gps_control.cpp

```diff
@@ -158,12 +158,7 @@ void Ekf::controlGpsFusion()
 			if (starting_conditions_passing) {
 				// Do not use external vision for yaw if using GPS because yaw needs to be
 				// defined relative to an NED reference frame
-				if (_control_status.flags.ev_yaw
-				    || _mag_inhibit_yaw_reset_req
-				    || _mag_yaw_reset_req) {
```

Comment:
If a mag yaw reset is pending, it's probably a sign that the heading isn't valid. Should we rather tell this through `yaw_align` ?
Also, if the mag fusion is inhibited for a long period of time, we don't know how good the yaw estimate is. The yaw estimate is quite critical for GNSS fusion so we need to make sure that we don't start the GNSS fusion if the yaw estimate is unaided.

Reply:
It didn't make the cut here, but in one of the EV branches I have a yaw estimator check (and possible yaw reset) here just before starting GPS.